packages/typography/src/Typography.tsx

```diff
@@ -102,22 +105,73 @@ function Body({ children, className, weight = 'regular', ...rest }: BodyProps) {
   );
 }
 
+const anchorDataProp = createDataProp('anchor-inline-code');
+
 const code = css`
+  background-color: ${uiColors.gray.light3};
+  border: 1px solid ${uiColors.gray.light1};
+  border-radius: 3px;
   font-family: ${fontFamilies.code};
-  display: inline-block;
+
+  ${anchorDataProp.selector}:hover > code > & {
+    border-color: ${uiColors.blue.base};
+  }
+`;
+
+const codeFocus = css`
+  ${anchorDataProp.selector}:focus > code > & {
+    border-color: ${uiColors.blue.base};
+    background-color: ${uiColors.gray.light3};
+  }
+`;
+
+const codeLink = css`
+  text-decoration: none;
+  margin: 0;
+  padding: 0;
+
+  &:focus {
+    outline: none;
+  }
+`;
+
+const nowrap = css`
+  white-space: nowrap;
 `;
 
-type InlineCodeProps = HTMLElementProps<'code'>;
+const normal = css`
+  white-space: normal;
+`;
+
+const colorBlue = css`
+  color: ${uiColors.blue.base};
+`;
+
+type InlineCodeProps = OneOf<HTMLElementProps<'code'>, HTMLElementProps<'a'>>;
 
 function InlineCode({ children, className, ...rest }: InlineCodeProps) {
+  const { usingKeyboard: showFocus } = useUsingKeyboardContext();
   const size = useBaseFontSize();
-  const body = size === 16 ? typeScale2 : typeScale1;
-
-  return (
-    <code {...rest} className={cx(sharedStyles, code, body, className)}>
-      {children}
+  const fontSize = size === 16 ? typeScale2 : typeScale1;
+  const whiteSpace =
+    typeof children === 'string' && children.length <= 30 ? nowrap : normal;
```

Comment:
I don't think this will have the expected behavior with unicode characters that have multiple code points. I think we want something like this: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/length#Unicode